src/spineml/simulator/simulator.cc

```diff
@@ -0,0 +1,820 @@
+: Simulator()
+{
+    load(experimentXML, overrideOutputPath);
+}
+//----------------------------------------------------------------------------
+Simulator::~Simulator()
+{
+     // Close model library if loaded successfully
+    if(m_ModelLibrary) {
+#ifdef _WIN32
+        FreeLibrary(m_ModelLibrary);
+#else
+        dlclose(m_ModelLibrary);
+#endif
+    }
+}
+//----------------------------------------------------------------------------
+void Simulator::load(const std::string &experimentXML, const std::string &overrideOutputPath)
+{
+    // Use filesystem library to get parent path of the network XML file
+    const auto experimentPath = filesystem::path(experimentXML).make_absolute();
+    const auto basePath = experimentPath.parent_path();
+
+    // If 2nd argument is specified use as output path otherwise use SpineCreator-compliant location
+    const filesystem::path outputPath = overrideOutputPath.empty() ? basePath.parent_path() : filesystem::path(overrideOutputPath).make_absolute();
+
+    LOGI << "Output path:" << outputPath.str();
+
+    // Load experiment document
+    pugi::xml_document experimentDoc;
+    auto experimentResult = experimentDoc.load_file(experimentPath.str().c_str());
+    if(!experimentResult) {
+        throw std::runtime_error("Unable to load experiment XML file:" + experimentPath.str() + ", error:" + experimentResult.description());
+    }
+
+    // Get SpineML root
+    auto experimentSpineML = experimentDoc.child("SpineML");
+    if(!experimentSpineML) {
+        throw std::runtime_error("XML file:" + experimentPath.str() + " is not a SpineML experiment - it has no root SpineML node");
+    }
+
+    // Get experiment node
+    auto experiment = experimentSpineML.child("Experiment");
+    if(!experiment) {
+            throw std::runtime_error("No 'Experiment' node found");
+    }
+
+    // Get model
+    auto model = experiment.child("Model");
+    if(!model) {
+        throw std::runtime_error("No 'Model' node found in experiment");
+    }
+
+    // Build path to network from URL in model
+    auto networkPath = basePath / model.attribute("network_layer_url").value();
+    LOGI << "Experiment using model:" << networkPath;
+
+    // Get the filename of the network and remove extension
+    // to get something usable as a network name
+    std::string networkName = networkPath.filename();
+    networkName = networkName.substr(0, networkName.find_last_of("."));
+
+    // Attempt to load model library
+#ifdef _WIN32
+    auto libraryPath = outputPath / "run" / "runner_Release.dll";
+    LOGI << "Experiment using model library:" << libraryPath;
+    m_ModelLibrary = LoadLibrary(libraryPath.str().c_str());
+#else
+    auto libraryPath = outputPath / "run" / (networkName + "_CODE") / "librunner.so";
+    LOGI << "Experiment using model library:" << libraryPath;
+    m_ModelLibrary = dlopen(libraryPath.str().c_str(), RTLD_NOW);
+#endif
```

Comment:
I'd personally be inclined to wrap the library loading etc. into its own class and header file to reduce the if-deffery in this file, but it's up to you

Reply:
Yeah - that's a future project. There's very similar logic here, in PyGeNN and in BoB robotics now